packages/netlify-cms-core/src/actions/__tests__/config.spec.js

```diff
@@ -101,5 +101,55 @@ describe('config', () => {
         ]),
       ).toEqual('_');
     });
+
+    it('should default public_folder to media_folder', () => {
+      expect(
+        applyDefaults(
+          fromJS({
+            collections: [{ folder: 'foo', media_folder: 'static/images/docs' }],
+          }),
+        ).get('collections'),
+      ).toEqual(
+        fromJS([
+          {
+            folder: 'foo',
+            media_folder: 'static/images/docs',
+            public_folder: 'static/images/docs',
+          },
+        ]),
+      );
+    });
+
+    it('should not override public_folder with media_folder', () => {
+      expect(
+        applyDefaults(
+          fromJS({
+            collections: [
+              { folder: 'foo', media_folder: 'static/images/docs', public_folder: 'images/docs' },
+            ],
+          }),
+        ).get('collections'),
+      ).toEqual(
+        fromJS([
+          {
+            folder: 'foo',
+            media_folder: 'static/images/docs',
+            public_folder: 'images/docs',
+          },
+        ]),
+      );
+    });
+
+    it("should set media_folder and public_folder to an empty string when path exists, but media_folder doesn't", () => {
+      expect(
+        applyDefaults(
+          fromJS({
+            collections: [{ folder: 'foo', path: '{{slug}}/index' }],
+          }),
+        ).get('collections'),
+      ).toEqual(
+        fromJS([{ folder: 'foo', path: '{{slug}}/index', media_folder: '', public_folder: '' }]),
+      );
+    });
```

Comment:
We should add some tests to ensure collection level media_folder and public_folder default to the global values when not set.

Reply:
That logic is actually part of the selectors:
https://github.com/netlify/netlify-cms/blob/476f45096efa1723936a73f2e2e04d5c7ccd293f/packages/netlify-cms-core/src/reducers/entries.ts#L148
https://github.com/netlify/netlify-cms/blob/476f45096efa1723936a73f2e2e04d5c7ccd293f/packages/netlify-cms-core/src/reducers/entries.ts#L193

And tested here:
https://github.com/netlify/netlify-cms/blob/476f45096efa1723936a73f2e2e04d5c7ccd293f/packages/netlify-cms-core/src/reducers/__tests__/entries.spec.js#L80
and here:
https://github.com/netlify/netlify-cms/blob/476f45096efa1723936a73f2e2e04d5c7ccd293f/packages/netlify-cms-core/src/reducers/__tests__/entries.spec.js#L185

Collection based `media_folder` folders values are relative to the entry path so they have different semantics than the global ones

Reply:
Ah, thanks for the pointers, sounds good 👍